docs/content/latest/api/ysql/datatypes/type_datetime/formatting-functions.md

```diff
@@ -0,0 +1,1193 @@
+
+```plpgsql
+with c as (
+  select
+    '2021-06-15'::date as d,
+    'dd-Mon-yyyy'      as fmt)
+select
+  to_char(d,              fmt) as "implicit cast to timestamptz",
+  to_char(d::timestamptz, fmt) as "explicit cast to timestamptz",
+  to_char(d::timestamp,   fmt) as "explicit cast to plain timestamp"
+from c;
+```
+
+This is the result:
+
+```output
+ implicit cast to timestamptz | explicit cast to timestamptz | explicit cast to plain timestamp 
+------------------------------+------------------------------+----------------------------------
+ 15-Jun-2021                  | 15-Jun-2021                  | 15-Jun-2021
+```
+
+Now try this:
+
+```plpgsql
+with c as (
+  select
+    '15:00:00'::time as t,
+    'hh12-mi-ss AM'  as fmt)
+select
+  to_char(t,           fmt) as "implicit cast to interval",
+  to_char(t::interval, fmt) as "explicit cast to interval"
+from c;
+```
+
+This is the result:
+
+```output
+ implicit cast to interval | explicit cast to interval 
+---------------------------+---------------------------
+ 03-00-00 PM               | 03-00-00 PM
+```
+
+You might think that it's strange that an _interval_ value can be rendered with _AM_ or _PM_. But it is what it is.
+
+```plpgsql
+select to_char(make_interval(hours=>15), 'AM');
+```
+
+This is the result:
+
+```output
+ PM
+```
+
+{{< tip title="Prefer uncluttered code to explicit typecasting with to_char() for date and time values." >}}
+
+Usually, users avoid writing code that brings an implicit typecast and prefer, instead, to write the typecast that they want explicitly. The tests above show that in the case that _to_char()_ is to be used to render a _date_ value or a _time_ value, the result is the same when the implicit typecast (respectively to _timestamptz_ or to _interval_) is allowed and when the explicit typecast is written. More carefully stated, writing the explicit typecast of a _date_ value either to a plain _timestamp_ value or to a _timestamptz_ value has the same effect as not writing a typecast; and writing the explicit typecast of a _time_ value to an _interval_ value has the same effect as not writing the typecast.
+
+Yugabyte recommends that, for the special case of invoking _to_char()_ with a _date_ or _time_ value, simply to allow the implicit typecast to have its innocent effect. In other words, you should prefer uncluttered code, here, to fastidious adherence to the normal rule of good practice.
```

Comment:
```suggestion
For the special case of invoking _to_char()_ with a _date_ or _time_ value, simply allow the implicit typecast to have its innocent effect. In other words, you should prefer uncluttered code, here, to fastidious adherence to the normal rule of good practice.
```

Reply:
I commented elsewhere that there are _47_ occurrences of _"Yugabyte recommends"_ within the _ysql_ tree. I won't change this.